tests/Resources/ti.ui.webview.addontest.js

```diff
@@ -0,0 +1,72 @@
+/*
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2011-Present by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+/* eslint-env mocha */
+/* global Ti */
+/* eslint no-unused-expressions: "off" */
+'use strict';
+var should = require('./utilities/assertions');
+
+describe('Ti.UI.WebView', function () {
+	var win,
+		didFocus = false;
+	this.slow(2000);
+	this.timeout(10000);
+
+	beforeEach(function () {
+		didFocus = false;
+	});
+
+	afterEach(function () {
+		if (win) {
+			win.close();
+		}
+		win = null;
+	});
+
+	it.iosMissing('#evalJS(string, function) - async variant', function (finish) {
+		var webview,
+			hadError = false;
+		win = Ti.UI.createWindow({
+			backgroundColor: 'blue'
+		});
+
+		webview = Ti.UI.createWebView();
+
+		webview.addEventListener('load', function () {
+			if (hadError) {
+				return;
+			}
+
+			// FIXME: Android is dumb and assumes no trailing semicolon!
+			webview.evalJS('Ti.API.info("Hello, World!");"WebView.evalJS.TEST"', function (result) {
```

Comment:
The Android implementation of evalJS leaves much to be desired. For one thing, it takes your input and appends:
```js
+"";
```

So if you have a trailing semicolon (as you should!) it doesn't behave properly. Also, I think it may be double-escaping strings here? I'd expect the result to be ```'WebView.evalJS.TEST'``` not ```'"WebView.evalJS.TEST"'```

Reply:
Welcome to Android. This is par for the course. Nothing is ever easy.